backend/app/controllers/spree/admin/base_controller.rb

```diff
@@ -62,6 +62,14 @@ def lock_order
       def order_mutex_redirect_path
         edit_admin_order_path(@order)
       end
+
+      def resource_not_found(options = {})
+        options[:custom_class] ||= model_class
+        options[:custom_redirect] ||= admin_url
+        flash[:error] = flash_message_for(options[:custom_class].new, :not_found)
+        redirect_to options[:custom_redirect]
+      end
```

Comment:
Refactored to use keyword arguments. I also don't think `custom_class` is a very good name for this argument because it doesn't tell you anything about what the argument is to be used for. I'm not sure what a better name would be though.

```suggestion
      def resource_not_found(custom_class: model_class, redirect_url: admin_url)
        flash[:error] = flash_message_for(custom_class.new, :not_found)
        redirect_to redirect_url
      end
```

Reply:
This makes much more sense, I will make the change. I tried to do this with the `#resourse_error_filter` (new name for the `nil_parent?`) but since I had to pass the options into `resource_not_found`, i found it difficult to accomplish. Perhaps you can have a look once I push the changes.